stdlib/3/smtplib.pyi

```diff
@@ -41,38 +45,42 @@ class SMTP:
     esmtp_features = ...  # type: Any
     source_address = ...  # type: Any
     local_hostname = ...  # type: Any
-    def __init__(self, host=..., port=..., local_hostname=..., timeout=...,
-                 source_address=...): ...
+    def __init__(self, host: str = ..., port: int = ...,
+                 local_hostname: Optional[str] = ..., timeout: Any = ...,
+                 source_address: Tuple[str, int] = ...) -> None: ...
     def __enter__(self): ...
     def __exit__(self, *args): ...
-    def set_debuglevel(self, debuglevel): ...
+    def set_debuglevel(self, debuglevel: int) -> None: ...
     sock = ...  # type: Any
     def connect(self, host=..., port=..., source_address=...): ...
     def send(self, s): ...
     def putcmd(self, cmd, args=...): ...
-    def getreply(self): ...
+    def getreply(self) -> _Reply: ...
     def docmd(self, cmd, args=...): ...
     def helo(self, name=...): ...
     def ehlo(self, name=...): ...
     def has_extn(self, opt): ...
     def help(self, args=...): ...
-    def rset(self): ...
-    def noop(self): ...
-    def mail(self, sender, options=...): ...
-    def rcpt(self, recip, options=...): ...
+    def rset(self) -> _Reply: ...
+    def noop(self) -> _Reply: ...
+    def mail(self, sender: str, options: List[str] = ...) -> _Reply: ...
+    def rcpt(self, recip: str, options: List[str] = ...) -> _Reply: ...
     def data(self, msg): ...
     def verify(self, address): ...
     vrfy = ...  # type: Any
     def expn(self, address): ...
     def ehlo_or_helo_if_needed(self): ...
     def login(self, user, password): ...
     def starttls(self, keyfile=..., certfile=..., context=...): ...
-    def sendmail(self, from_addr, to_addrs, msg, mail_options=...,
-                 rcpt_options=...): ...
-    def send_message(self, msg, from_addr=..., to_addrs=..., mail_options=...,
-                     rcpt_options=...): ...
+    def sendmail(self, from_addr: str, to_addrs: Union[str, List[str]],
+                 msg: Union[bytes, str], mail_options: List[str] = ...,
+                 rcpt_options: List[str] = ...) -> _SendErrs: ...
+    def send_message(self, msg: _Message, from_addr: str = ...,
```

Comment:
from_addr and to_addrs are allowed to be None, so they should be Optional. Also, a few more opportunities for using Sequence. `mail_options` should remain List though, because it does `mail_options += ['SMTPUTF8', 'BODY=8BITMIME']` and that only works on a List.